src/Avalonia.Controls/Primitives/SelectingItemsControl.cs

```diff
@@ -531,11 +531,28 @@ protected override void OnTextInput(TextInputEventArgs e)
 
                 _textSearchTerm += e.Text;
 
-                bool match(ItemContainerInfo info) =>
-                    info.ContainerControl is IContentControl control &&
-                    control.Content?.ToString()?.StartsWith(_textSearchTerm, StringComparison.OrdinalIgnoreCase) == true;
+                bool Match(ItemContainerInfo info)
+                {
+                    foreach (var child in info.ContainerControl.GetSelfAndVisualDescendants().OfType<Control>())
```

Comment:
Do I understand correctly, we need GetSelfAndVisualDescendants to get this property from the deep controls? 
Previously I though it's expected to only read root element. Is it WPF's behavior?
What expected behavior, when there are two levels of SelectingItemsControls with enabled text seach, but TextSearch.Text is defined only on inner list's data template?

Reply:
Because outer SelectingItemsControl will read TextSearch.Text of inner SelectingItemsControl's items with this code.